### ISSUE TITLE
Endless loop

### ISSUE THREAD

**Issue by EfiHerbst** [csmith-project/creduce#68]
Hi,

When running creduce on the following .c file, with the attached script, I get an endless loop after pass_clang :: local-to-global:
````
...
(-218.4 %, 6989 bytes)
(-219.3 %, 7009 bytes)
(-220.2 %, 7029 bytes)
(-221.1 %, 7049 bytes)
(-222.1 %, 7069 bytes)
(-223.0 %, 7089 bytes)
...
````
Test.sh
```
#!/bin/bash -e
gcc -m64 -Wall -O3  -c c_enc.c
nm c_enc.o | grep "U CAST_S_table0"
```
e_enc.c
```
# define CAST_LONG unsigned int
typedef struct cast_key_st                 CAST_KEY;
#if definedCAST_PTR
#else
# define E_CAST(n,key,L,R,OP1,OP2,OP3) \
        CAST_LONG a,b,c,d; \
        a=CAST_S_table0[t>> 8&0xff]; \
        b=CAST_S_table1[t    &0xff]; \
        c=CAST_S_table2[t>>24&0xff]; \
        d=CAST_S_table3[t>>16&0xff]; \
        L^=a OP2 b&0xffffffffL OP3 c&0xffffffffL OP1 d&0xffffffffL; \

#endif
extern const CAST_LONG CAST_S_table0[256];
                                extern const CAST_LONG CAST_S_table1[256];
                                extern const CAST_LONG CAST_S_table2[256];
                                extern const CAST_LONG CAST_S_table3[256];
                                void fn1(CAST_LONG *data, const CAST_KEY *key) {
                        CAST_LONG l, t;
                        E_CAST(1, k, l, r, -, +, ^) data[1] = l;
                        }
````

Thank you!

**Comment by regehr** [comment]
Ugh, thanks, we'll look into it.

**Comment by chenyang78** [comment]
Thanks for reporting the issue, @EfiHerbst 

Should be fixed in a5ead7f.

**Comment by EfiHerbst** [comment]
Thank you for the quick response, @regehr @chenyang78
